ingestion/tests/unit/profiler/pandas/test_custom_metrics.py

```diff
@@ -0,0 +1,233 @@
+    S3Config,
+)
+from metadata.generated.schema.entity.services.connections.database.datalakeConnection import (
+    DatalakeConnection,
+)
+from metadata.generated.schema.security.credentials.awsCredentials import AWSCredentials
+from metadata.generated.schema.tests.customMetric import CustomMetric
+from metadata.profiler.interface.pandas.profiler_interface import (
+    PandasProfilerInterface,
+)
+from metadata.profiler.processor.core import Profiler
+
+BUCKET_NAME = "MyBucket"
+
+
+@mock_s3
+class MetricsTest(TestCase):
+    """
+    Run checks on different metrics
+    """
+
+    current_dir = os.path.dirname(__file__)
+    resources_dir = os.path.join(current_dir, "resources")
+
+    datalake_conn = DatalakeConnection(
+        configSource=S3Config(
+            securityConfig=AWSCredentials(
+                awsAccessKeyId="fake_access_key",
+                awsSecretAccessKey="fake_secret_key",
+                awsRegion="us-west-1",
+            )
+        )
+    )
+
+    dfs = [
+        pd.read_csv(os.path.join(resources_dir, "profiler_test_.csv"), parse_dates=[5])
+    ]
+
+    table_entity = Table(
+        id=uuid4(),
+        name="user",
+        columns=[
+            EntityColumn(
+                name=ColumnName(__root__="id"),
+                dataType=DataType.INT,
+            )
+        ],
+    )
+
+    def setUp(self):
+        # Mock our S3 bucket and ingest a file
+        boto3.DEFAULT_SESSION = None
+        self.client = boto3.client(
+            "s3",
+            region_name="us-weat-1",
+        )
+
+        # check that we are not running our test against a real bucket
+        try:
+            s3 = boto3.resource(
+                "s3",
+                region_name="us-west-1",
+                aws_access_key_id="fake_access_key",
+                aws_secret_access_key="fake_secret_key",
+            )
+            s3.meta.client.head_bucket(Bucket=BUCKET_NAME)
+        except botocore.exceptions.ClientError:
+            pass
+        else:
+            err = f"{BUCKET_NAME} should not exist."
+            raise EnvironmentError(err)
+        self.client.create_bucket(
+            Bucket=BUCKET_NAME,
+            CreateBucketConfiguration={"LocationConstraint": "us-west-1"},
+        )
+
+        resources_paths = [
+            os.path.join(path, filename)
+            for path, _, files in os.walk(self.resources_dir)
+            for filename in files
+        ]
+
+        self.s3_keys = []
+
+        for path in resources_paths:
+            key = os.path.relpath(path, self.resources_dir)
+            self.s3_keys.append(key)
+            self.client.upload_file(Filename=path, Bucket=BUCKET_NAME, Key=key)
+
+        with patch.object(
+            PandasProfilerInterface,
+            "_convert_table_to_list_of_dataframe_objects",
+            return_value=self.dfs,
+        ):
+            self.sqa_profiler_interface = PandasProfilerInterface(
```

Comment:
can we add kwargs here instead of passing it without name? Not sure if we are actually supporting that on the interface itself

Reply:
what do you mean?

Reply:
sorry just to understand the calls a bit better, to have
```python
PandasProfilerInterface(
                param1=self.datalake_conn,
                param2=None,
                param3=table_entity,
                ...None,
                None,
                None,
                None,
                None,
                thread_count=1,
```